### ISSUE TITLE
kubeadm kube-controller-manager does not have ceph rbd binary anymore

### ISSUE THREAD

**Issue by saagievictor** [kubernetes/kubernetes#71904]
Hello :-)

**What happened**: 
I have updated my cluster from 1.11.3 to 1.12.3, thus converting hypercube use to kubeadm.
Everything is working but the creation of **new** RBD persistent Volume.
When I create a new volume, it fails continuously with the error:
```
Error: "failed to create rbd image: executable file not found in $PATH, command output: "
```
After some checks, it appears that the kube-controller-manager docker image which is used to create PV has no ceph rbd binary, while the older hybercube image had.

**What you expected to happen**:
The kubeadm kube-controller-manager image should be build with the ceph rbd binaries as the RBD provisioner is still officially supported?

**How to reproduce it (as minimally and precisely as possible)**:
* Create a storage class using ceph rbd:
```
allowVolumeExpansion: true
apiVersion: storage.k8s.io/v1
kind: StorageClass
metadata:
  annotations:
  creationTimestamp: null
  name: ceph-storageclass
parameters:
  adminId: admin
  adminSecretName: ceph-secret-admin
  adminSecretNamespace: default
  fsType: ext4
  imageFeatures: layering
  imageFormat: "2"
  monitors: 10.8.001:6789
  pool: common
  userId: common
  userSecretName: ceph-secret-common
provisioner: kubernetes.io/rbd
reclaimPolicy: Delete
volumeBindingMode: Immediate
```
* Create a pvc using this storageclass:
```
apiVersion: v1
kind: PersistentVolumeClaim
metadata:
  annotations:
    pv.kubernetes.io/bind-completed: "yes"
    pv.kubernetes.io/bound-by-controller: "yes"
    volume.beta.kubernetes.io/storage-provisioner: kubernetes.io/rbd
  finalizers:
  - kubernetes.io/pvc-protection
  name: test-rbd
spec:
  accessModes:
  - ReadWriteOnce
  dataSource: null
  resources:
    requests:
      storage: 5Gi
  storageClassName: ceph-storageclass
```

This one should works even without a running ceph cluster, as we only check the rbd binary, not create real image :-)

After that, the pv fails to create, and looking at the kube-controller-manager logs display:
```
I1210 07:20:57.485188       1 event.go:221] Event(v1.ObjectReference{Kind:"PersistentVolumeClaim", Namespace:"xxx-xxx-xxx", Name:"test-ceph", UID:"009be4ed-fa2e-11e8-9223-42010a0a005a", APIVersion:"v1", ResourceVersion:"5393206", FieldPath:""}): type: 'Warning' reason: 'ProvisioningFailed' Failed to provision volume with StorageClass "ceph-storageclass": failed to create rbd image: executable file not found in $PATH, command output: 
```

**Anything else we need to know?**:
I have made some test with older images from hyperkube, up to the 1.11.5 the rbd binary is present:
```
$ sudo docker run gcr.io/google-containers/hyperkube-amd64:v1.11.5  rbd --version
ceph version 10.2.5 (c461ee19ecbc0c5c330aca20f7392c9a00730367)
```
But with kubeadm, no more binary presents:
```
$ sudo docker run gcr.io/google-containers/kube-controller-manager:v1.12.3 rbd --version
docker: Error response from daemon: OCI runtime create failed: container_linux.go:348: starting container process caused "exec: \"rbd\": executable file not found in $PATH": unknown.
ERRO[0000] error waiting for container: context canceled 
```

I have tested some 1.12 beta images, and 1.13.0, same problem.

**Environment**:
- Kubernetes version (use `kubectl version`):
```
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"12", GitVersion:"v1.12.3", GitCommit:"435f92c719f279a3a67808c80521ea17d5715c66", GitTreeState:"clean", BuildDate:"2018-11-26T12:46:57Z", GoVersion:"go1.10.4", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"12", GitVersion:"v1.12.3", GitCommit:"435f92c719f279a3a67808c80521ea17d5715c66", GitTreeState:"clean", BuildDate:"2018-11-26T12:46:57Z", GoVersion:"go1.10.4", Compiler:"gc", Platform:"linux/amd64"}
```
- Cloud provider or hardware configuration: 1 master/3 nodes on GCE VM and 3 masters/3 etcd/35 nodes on HP hardware.
- OS (e.g. from /etc/os-release): `VERSION="16.04.4 LTS (Xenial Xerus)"`
- Kernel (e.g. `uname -a`): `Linux a4231cmnk8smaster1 4.15.0-1025-gcp #26~16.04.1-Ubuntu SMP Wed Nov 21 10:13:20 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux`
- Install tools: cluster installed and managerd through ansible with kubespray v2.8.0
- Others:

/kind bug

**Comment by neolit123** [comment]
> The kubeadm kube-controller-manager image should be build with the ceph rbd binaries as the RBD provisioner is still officially supported?

this is not a kubeadm problem directly but rather a release process issue. kubeadm is just a consumer of the images.

it's unclear to me why this binary was removed.

/sig release

**Comment by chdxD1** [comment]
Can confirm. Setting up a cluster and can't use rbd/ceph is a bit bad.

**Comment by shuraa** [comment]
Can confirm too. Transition to the 1.12 blocked by missing rbd binary in the $PATH.

**Comment by neolit123** [comment]
/close 

this a duplicate of:
https://github.com/kubernetes/kubernetes/issues/56990

see this comment:
https://github.com/kubernetes/kubernetes/issues/38923#issuecomment-315255075

no idea where this is in the changelog for 1.12.

**Comment by valumar** [comment]
Still have an issue in fresh installed k8s 1.16.3 by kubeadm tool:
sc.yml:
```
apiVersion: storage.k8s.io/v1
kind: StorageClass
metadata:
  name: rbd-data
provisioner: kubernetes.io/rbd
parameters:
  monitors: 10.x.x.1:6789, 10.x.x.2:6789, 10.x.x.3:6789
  adminId: kube
  adminSecretName: ceph-secret
  adminSecretNamespace: "kube-system"
  pool: rbd_data
  userId: kube
  userSecretName: ceph-secret-user
```
pvc.yml:
```
apiVersion: v1
kind: PersistentVolumeClaim
metadata:
  name: test-3gb
spec:
  accessModes:
    - ReadWriteOnce
  resources:
    requests:
      storage: 3Gi
  storageClassName: rbd-data

```

**Comment by 0x4139** [comment]
is there any plan to support the binaries inside the kube-controller-manager or we are stuck to using and external provisioner while the docs specify that the rbd provisioner is supported internaly?

**Comment by neolit123** [comment]
@0x4139 hi, discussion is happening here https://github.com/kubernetes/kubernetes/issues/85454
